config/v1/types_infrastructure.go

```diff
@@ -528,18 +528,21 @@ type AWSPlatformStatus struct {
 
 // AWSResourceTag is a tag to apply to AWS resources created for the cluster.
 type AWSResourceTag struct {
-	// key is the key of the tag
+	// key sets the key of the AWS resource tag key-value pair. Key is required when defining an AWS resource tag.
+	// Length Constraints: Minimum length of 1. Maximum length of 128.
+	// The string can contain only the set of Unicode letters, digits, white-space, '_', '.', '/', '=', '+', '-', ':', '@' and cannot be prefixed with 'aws:'.
```

Comment:
The validation pattern no longer looks like it technically supports all unicode letters, digits, and whitespace. Maybe something like:
```suggestion
	// The string can contain only the set of alphanumeric characters, space (' '), '_', '.', '/', '=', '+', '-', ':', '@' and cannot be prefixed with 'aws:'.
```
?

This will likely also need to be updated further depending on any changes related to the prefix ratcheting.